File: aoa_server/game/Player.cpp
#include <string>


#include "Player.h"
#include "../core/Logger.h"


Player::Player() {}

Player::Player(int id) {
    this->ID = id;
    this->init();
}

Player::Player(int id, std::string username) {
    this->ID = id;
    this->username = username;
    this->init();
}

void Player::init() {
    this->setRoomID(-1);
    this->setOnline();
    this->clearIncorrectMsgCount();
    this->setUsername("");
}

void Player::setID(int id) {
    this->ID = id;
}

int Player::getID() const {
    return this->ID;
}
void Player::setRoomID(int id) {
    this->roomID = id;
}

int Player::getRoomID() const {
    return this->roomID;
}

void Player::setUsername(std::string username) {
    this->username = username;
}

std::string Player::getUsername() const {
    return this->username;
}

bool Player::hasUsername() const {
    return this->username.length() > 0;
}

bool Player::hasRoom() {
    return this->roomID != -1;
}

void Player::leaveRoom() {
    this->setRoomID(-1);
}

void Player::setStatus(bool online) {
    this->online = online;
}

void Player::setOnline() {
    this->setStatus(true);
}

void Player::setOffline() {
    this->setStatus(false);
}

bool Player::isOnline() const {
    bool on = this->online;
     return on;
}

int Player::getIncorrectMsgCount() const {
    return incorrectMsgCount;
}

void Player::registerIncorrectMsgCount() {
    this->incorrectMsgCount++;
}

void Player::clearIncorrectMsgCount() {
    this->incorrectMsgCount = 0;
}

void Player::merge(Player *p) {
    this->setRoomID(p->getRoomID());
    this->setOnline();
    this->clearIncorrectMsgCount();
    this->setUsername(p->getUsername());
}

File: aoa_client_mvn/src/main/java/cz/kiv/ups/model/Player.java
package cz.kiv.ups.model;

public class Player {
	
	private int ID = -1;
	
	private String name;

	private boolean isActive;

	private boolean isOnline;

	private int roomID = -1;

	// playerlist
	public Player(int id, String username, boolean online, boolean active) {
		this.ID = id;
		this.name = username;
		this.isOnline = online;
		this.isActive = active;
	}

	public Player(String name) {
		this.name = name;
	}
	
	public Player(int id, String name) {
		this.ID = id;
		this.name = name;
	}
	
	
	
	
	




	/**
	 * @return the ID
	 */
	public int getID() {
		return ID;
	}

	/**
	 * @param id the ID to set
	 */
	public void setID(int id) {
		ID = id;
	}
	
	public boolean hasID(){
		return this.ID != -1;
	}

	/**
	 * @return the name
	 */
	public String getName() {
		return name;
	}

	/**
	 * @param name the name to set
	 */
	public void setName(String name) {
		this.name = name;
	}

	/**
	 * @return the isOnline
	 */
	public boolean isOnline() {
		return isOnline;
	}

	/**
	 * @param online the isOnline to set
	 */
	public void setOnline(boolean online) {
		this.isOnline= online;
	}

    /**
     *
     */
    public boolean isActive() {
        return this.isActive;
    }

    /**
     *
     */
    public void setActive(boolean active) {
        this.isActive = active;
    }

	public static Player[] parsePlayers(String sequence, String delimiter){
		String[] parts = sequence.split(delimiter);
		Player[] players = new Player[parts.length];
		
		for (int i = 0; i < parts.length; i++) {
			players[i] = new Player(parts[i]);
		}
		
		return players;
	}

    public boolean hasRoom() {
        return this.roomID != -1;
    }

    public int getRoomID() {
        return this.roomID;
    }

    public void setRoomID(int roomID) {
        this.roomID = roomID;
    }

    public String toString(){
		return "user(" + this.ID + " | " + this.name + ")";
	}
}
